Sources/SpeziScheduler/Task/Event.swift

```diff
@@ -90,57 +97,58 @@ public struct Event {
         self.occurrence = occurrence
         self.outcomeState = State(outcome)
     }
+}
 
-    /// Complete the event.
-    ///
-    /// Does nothing if the event is already completed.
-    @MainActor
-    @discardableResult
-    public func complete() -> Outcome {
-        self.complete { _ in }
+
+extension Event {
+    public enum CompletionError: Error {
+        /// The event's ``AllowedCompletionPolicy`` doesn't allow completing the event at this time.
+        case preventedByCompletionPolicy
     }
     
     /// Complete the event with additional information.
     ///
-    /// ```swift
-    /// var event: Event
+    /// - parameter ignoreCompletionPolicy: Allows for forced completion of the event, even if the underlying task's ``AllowedCompletionPolicy`` would otherwise prohibit it. Defaults to `false`.
+    /// - parameter closure: A closure that allows setting properties of the outcome.
     ///
-    /// event.complete {
-    ///     event.myCustomData = "..."
+    /// ```swift
+    /// event.complete { outcome in
+    ///     outcome.myCustomData = "..."
     /// }
     /// ```
     ///
     /// - Warning: If the event is already completed, the closure will be applied to the existing outcome.
-    ///
-    /// - Parameter closure: A closure that allows setting properties of the outcome.
     @MainActor
     @discardableResult
-    public func complete(with closure: (Outcome) -> Void) -> Outcome {
+    public func complete(
```

Comment:
If we make this PR a breaking change I would be fine with the additional thrown error as the only option. If we want to avoid a breaking change we should probably introduce a method that just catches the error and mark it as deprecated to push people to the newer version?

Reply:
IMO we could get away with not making it a breaking change (the functionality is a superset of what existed before), and if we add a couple overloads it should work without any issues